Source/Core/InputCommon/ControllerInterface/WGInput/WGInput.h

```diff
@@ -0,0 +1,225 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#pragma warning(push)
+// Disable warning for virtual functions with no virtual destructor:
+#pragma warning(disable : 4265)
+#include <wrl.h>
+#pragma warning(pop)
+
+#include <windows.gaming.input.h>
+
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+
+namespace ciface
+{
+namespace WGInput
+{
+namespace WGI = ABI::Windows::Gaming::Input;
+using Microsoft::WRL::ComPtr;
+
+void Init();
+void DeInit();
+void PopulateDevices();
+
+class Device : public Core::Device
+{
+protected:
+  class Button : public Core::Device::Input
+  {
+  public:
+    explicit Button(const boolean* _button) : m_button(*_button) {}
+    ControlState GetState() const override;
+
+  private:
+    // boolean comes from Windows API (typedef of unsigned char)
+    const boolean& m_button;
+  };
+
+  // A button with one of the "labels" that WGI provides:
+  class NamedButton : public Button
+  {
+  public:
+    NamedButton(const boolean* _button, const char* _name) : Button(_button), m_name(_name) {}
```

Comment:
```suggestion
    NamedButton(const boolean* button, const char* name) : Button(button), m_name(name) {}
```

etc for relevant constructors of other types below